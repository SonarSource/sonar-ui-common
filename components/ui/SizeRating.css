/*
 * Sonar UI Common
 * Copyright (C) 2019-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
.size-rating {
  display: inline-block;
  vertical-align: top;
  width: var(--controlHeight);
  height: var(--controlHeight);
  line-height: var(--controlHeight);
  border-radius: var(--controlHeight);
  background-color: var(--blue);
  color: #fff;
  font-size: var(--smallFontSize);
  text-align: center;
  text-shadow: 0 0 1px rgba(0, 0, 0, 0.35);
}

.size-rating-small {
  width: 18px;
  height: 18px;
  line-height: 18px;
  margin-top: -1px;
  margin-bottom: -1px;
  font-size: 10px;
}

.size-rating-muted {
  background-color: #bdbdbd;
}
