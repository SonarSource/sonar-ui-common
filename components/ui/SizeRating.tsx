/*
 * Sonar UI Common
 * Copyright (C) 2019-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as classNames from 'classnames';
import { inRange } from 'lodash';
import * as React from 'react';
import './SizeRating.css';

export interface Props {
  muted?: boolean;
  small?: boolean;
  value: number | null | undefined;
}

export default function SizeRating({ small = false, muted = false, value }: Props) {
  if (value == null) {
    return <div className="size-rating size-rating-muted">&nbsp;</div>;
  }

  let letter;
  if (inRange(value, 0, 1000)) {
    letter = 'XS';
  } else if (inRange(value, 1000, 10000)) {
    letter = 'S';
  } else if (inRange(value, 10000, 100000)) {
    letter = 'M';
  } else if (inRange(value, 100000, 500000)) {
    letter = 'L';
  } else if (value >= 500000) {
    letter = 'XL';
  }

  const className = classNames('size-rating', {
    'size-rating-small': small,
    'size-rating-muted': muted
  });

  return (
    <div aria-hidden="true" className={className}>
      {letter}
    </div>
  );
}
