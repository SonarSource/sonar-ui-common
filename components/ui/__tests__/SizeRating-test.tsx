/*
 * Sonar UI Common
 * Copyright (C) 2019-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import SizeRating, { Props } from '../SizeRating';

it('should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot();
  expect(shallowRender({ muted: true, small: true, value: 1000 })).toMatchSnapshot();
  expect(shallowRender({ value: 10000 })).toMatchSnapshot();
  expect(shallowRender({ value: 100000 })).toMatchSnapshot();
  expect(shallowRender({ value: 500000 })).toMatchSnapshot();
});

function shallowRender(props: Partial<Props> = {}) {
  return shallow(<SizeRating value={100} {...props} />);
}
