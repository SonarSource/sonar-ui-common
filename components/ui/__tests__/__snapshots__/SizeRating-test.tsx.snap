// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  aria-hidden="true"
  className="size-rating"
>
  XS
</div>
`;

exports[`should render correctly 2`] = `
<div
  aria-hidden="true"
  className="size-rating size-rating-small size-rating-muted"
>
  S
</div>
`;

exports[`should render correctly 3`] = `
<div
  aria-hidden="true"
  className="size-rating"
>
  M
</div>
`;

exports[`should render correctly 4`] = `
<div
  aria-hidden="true"
  className="size-rating"
>
  L
</div>
`;

exports[`should render correctly 5`] = `
<div
  aria-hidden="true"
  className="size-rating"
>
  XL
</div>
`;
